build.Containerfile

```diff
@@ -0,0 +1,39 @@
+#
+# Reproducible reference build
+#
+# Usage:
+#
+# docker build --file build.Containerfile --output <outputdir> .
+# or
+# podman build --file build.Containerfile --output <outputdir> .
+#
+# The build artifacts are written to the specified output directory.
+#
+
+FROM debian:bullseye-slim AS build-stage
+
+# prepare debian environment
+ENV DEBIAN_FRONTEND noninteractive
+RUN /usr/bin/apt-get update && \
+    /usr/bin/apt-get --yes install openjdk-11-jdk-headless gradle && \
+    /usr/sbin/adduser --disabled-login --gecos "" builder
+
+# give up privileges
+USER builder
+WORKDIR /home/builder
+
+# copy project source code
+COPY --chown=builder / project/
+
+# build project
+RUN /usr/bin/gradle --project-dir project/ \
+    --no-build-cache --no-daemon --no-parallel \
+    -Dmaven.repo.local=repo \
+    clean :bitcoinj-core:publishToMavenLocal :bitcoinj-wallettool:installDist
```

Comment:
1. it's for getting the POM with a nice name
2. the structure is close resembles what we upload to Maven Central, so I thought it's a good fit

Reply:
Does this answer your question?

Reply:
It answers my question, but I'd prefer if we just used the direct outputs of the primary build targets as the reproducible outputs.

Reply:
To avoid making an unnecessary copy of the files and to focus our verification efforts on the normal build outputs.

And there's a philosophical (and possibly practical) objection. Publishing to maven local is technically a side effect and mutates the developers system.  (This is in the context of "functional-style" build tools like Nix and Guix.)  I know you have it configured to write to the local directory, but using `publishToMavenLocal` just "feels wrong" and makes me nervous. This may just be superstition on my part, though.

Reply:
By the way, the "publish" tasks don't run our tests. This can be seen as bad but also as good. In fact, reproducable builds is *not* about correctness of code, it's just about reproducibility. Since we have tons of tests for correctness, I thought not running the tests in our reproducable build is nice to have.

However, I'm not adamant on omitting the tests. We could add the `:bitcoinj-core:build` or `:bitcoinj-core:test` tasks to the gradle line. However, I'm pretty sure I will not be able to verify on my Raspi Zero 2W any more if we do…

Reply:
> I think we have some mismatch of understanding of what reproducible builds (or maybe verifyable builds?) is about. 

I don't think so. I think we just differ over what should be the final output.

I would like a regular. local build to produce output that is reproducible and to be able to easily verify the shasums. We pretty much have that now, so maybe this doesn't matter.

Reply:
> By the way, the "publish" tasks don't run our tests. 

I noticed this. I suppose I would feel better if the tests were run as part of the process of producing an official image, but I suppose the tests could be run separately and the shasums of the two images compared so that we know the official image passed the tests.

Perhaps there could be a config flag to turn on/off testing in the reference build?